source/collections/Vector2D.ooc

```diff
@@ -38,6 +40,58 @@ Vector2D: class <T> {
 	_elementPosition: func (row, column: Int, columnCount := this columnCapacity) -> Int {
 		result := columnCount * row + column
 	}
+	resize: func (newRowCapacity, newColumnCapacity: Int) {
+		if (newColumnCapacity == this columnCapacity)
+			if (newRowCapacity == this rowCapacity)
+				return
+			else
+				_allocate(newRowCapacity, newColumnCapacity)
+		else {
+			temporaryResult: T*
+			minimumRowCapacity := Int minimum(this rowCapacity, newRowCapacity)
+			minimumColumnCapacity := Int minimum(this columnCapacity, newColumnCapacity)
+
+			if (newColumnCapacity > this columnCapacity && newRowCapacity > this rowCapacity)
+				temporaryResult = gc_calloc(newRowCapacity * newColumnCapacity, T size)
+			else
+				temporaryResult = gc_malloc(newRowCapacity * newColumnCapacity * T size)
+
+			for (row in 0 .. minimumRowCapacity)
+				memcpy(temporaryResult[T size * this _elementPosition(row, 0, newRowCapacity)]&,
+					this _backend[T size * this _elementPosition(row, 0)]&, minimumColumnCapacity * T size)
+
+			this free()
+			this init(temporaryResult, newRowCapacity, newColumnCapacity)
```

Comment:
I don't recall ever seeing anything free itself and then run its own `init` function afterwards, and I hadn't even considered it a possibility. It's not a problem at all!